backend/app/usecase/search/order/created_time.go

```diff
@@ -0,0 +1,43 @@
+package order
+
+import (
+	"sort"
+
+	"github.com/short-d/short/backend/app/entity"
+)
+
+var _ Order = (*CreatedTime)(nil)
+
+// CreatedTime arranges searchable resources based on their created time.
+type CreatedTime struct {
+}
+
+// ArrangeShortLinks arranges shortLinks based on CreatedAt.
+// The returned short links satisfies the created before comparision if CreatedAt
+// is not nil, otherwise it sits on the end of the slices if CreatedAt is nil.
+func (c CreatedTime) ArrangeShortLinks(shortLinks []entity.ShortLink) []entity.ShortLink {
+	sort.SliceStable(shortLinks, func(i, j int) bool {
+		if shortLinks[j].CreatedAt == nil {
+			return true
+		} else if shortLinks[i].CreatedAt == nil {
+			return false
+		}
+		return shortLinks[i].CreatedAt.Before(*shortLinks[j].CreatedAt)
+	})
+	return shortLinks
+}
+
+// ArrangeUsers arranges users based on CreatedAt.
+// The returned users satisfies the created before comparision if CreatedAt
+// is not nil, otherwise it sits on the end of the slices if CreatedAt is nil.
+func (c CreatedTime) ArrangeUsers(users []entity.User) []entity.User {
+	sort.SliceStable(users, func(i, j int) bool {
+		if users[j].CreatedAt == nil {
+			return true
+		} else if users[i].CreatedAt == nil {
+			return false
+		}
+		return users[i].CreatedAt.Before(*users[j].CreatedAt)
+	})
```

Comment:
I think the internal implementation of `ArrangeShortLinks` can be extracted out into a private function being reuse between itself and `ArrangeUsers`.

Reply:
I will try to refactor!